### PR TITLE
Update endpoint filters

### PR DESCRIPTION
**Pull request by jefmoura**
## Purpose
We have overwritten the list method but we don't use the `filter_queryset` method in it, so we had to update the queryset of all endpoints.

### Consideration
- Some unit tests were updated to cover more cases and have a standard.
- Some `filter_fields` were removed because they don't exist in the models anymore.

Related issue: #802

### CODE REVIEW

File: feed/views.py [hunk 1]
```diff
@@ -1217,21 +1185,19 @@ def list(self, request):
 class MilestoneViewSet(viewsets.ModelViewSet):
 
     def list(self, request):
-        if request.user.is_superuser:
-            queryset = self.queryset
-        else:
+        # Use this queryset or the django-filters lib will not work
```
Comment (Menda) [inline feed/views.py]: I think this comment can be removed from every function. You are a mature and trusted developer, and if I doubted why you are using the filter_queryset function then I use the PR review to ask you.

File: feed/tests/test_documentationview.py [hunk 2]
```diff
@@ -206,3 +218,83 @@ def test_delete_documentation_normal_user(self):
...
+            organization=self.tola_user.organization)
+        wflvl2_2 = factories.WorkflowLevel2(workflowlevel1=wflvl1_2)
+        WorkflowTeam.objects.create(workflow_user=self.tola_user,
+                                    workflowlevel1=wflvl1_2)
+        factories.Documentation(name='Document 2', workflowlevel1=wflvl1_2,
+                                workflowlevel2=wflvl2_2)
+
+        request = self.factory.get(
+            '/api/documentation'
+            '/?workflowlevel2__workflowlevel1__country__country=%s' %
```
Comment (Menda) [inline feed/tests/test_documentationview.py]: It's a pity that someone needs to know so much about the internal details and relations of a model to filter properly in the API. But yeah, that's life...

File: feed/views.py (hunk 1 above)
Comment (jefmoura) [reply feed/views.py]: I also think I could remove the comments but the problem is: when another developer gets the code and changes it back as it was before. I wanna know from you: Is a short warning comment in the code useless?

File: feed/tests/test_documentationview.py (hunk 2 above)
Comment (jefmoura) [reply feed/tests/test_documentationview.py]: I think its possible to change but we need to create FilterSet for them. We check it later.

File: feed/views.py (hunk 1 above)
Comment (Menda) [reply feed/views.py]: With "this queryset" you mean `self.get_queryset()`?

Comment (jefmoura) [reply feed/views.py]: No. I meant the whole line: `queryset = self.filter_queryset(self.get_queryset())`

**Pull request merged**